core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/dictionary/DictionaryMatcherOpDesc.scala

```diff
@@ -0,0 +1,53 @@
+package edu.uci.ics.texera.workflow.operators.dictionary
+
+import com.fasterxml.jackson.annotation.{JsonProperty, JsonPropertyDescription}
+import com.google.common.base.Preconditions
+import edu.uci.ics.texera.workflow.common.metadata.annotations.AutofillAttributeName
+import edu.uci.ics.texera.workflow.common.metadata.{
+  InputPort,
+  OperatorGroupConstants,
+  OperatorInfo,
+  OutputPort
+}
+import edu.uci.ics.texera.workflow.common.operators.{OneToOneOpExecConfig, OperatorDescriptor}
+import edu.uci.ics.texera.workflow.common.tuple.schema.{AttributeType, OperatorSchemaInfo, Schema}
+
+/**
+  * Dictionary matcher operator matches a tuple if the specified column is in the given dictionary.
+  * It outputs an extra column to label the tuple if it is matched or not
+  * This is the description of the operator
+  */
+class DictionaryMatcherOpDesc extends OperatorDescriptor {
+  @JsonProperty(value = "Dictionary", required = true)
+  @JsonPropertyDescription("dictionary values separated by a comma") var dictionary: String = _
+
+  @JsonProperty(value = "Attribute", required = true)
+  @JsonPropertyDescription("column name to match")
+  @AutofillAttributeName var attribute: String = _
+
+  @JsonProperty(value = "result attribute", required = true, defaultValue = "matched")
+  @JsonPropertyDescription("column name of the matching result") var resultAttribute: String = _
+
+  @JsonProperty(value = "Matching type", required = true) var matchingType: MatchingType = _
```

Comment:
an alternative design would be just filtering out the values, just curious why do you prefer adding a separate column over filtering?

Reply:
that was the original implementation, but the user wants an extra column to label a tuple if it is matched but not filter it

Reply:
got it